### PR TITLE
Modify update_shipment_handler

### PR DESCRIPTION
**Pull request by philbirt**
* If the shipment is transitioning to shipped, then we want to call OrderShipping directly instead of the state machine transition

We are creating Cartons to represent shipped shipments. We would like to remove all references to `Shipment.ship!` and call our new `OrderShipping` class directly. Cartons PR: https://github.com/bonobos/spree/pull/276

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'spree', github: 'bonobos/spree', branch: '2-2-dev'
+gem 'spree', github: 'bonobos/spree', branch: 'cartons-step-1'
```
Comment (philbirt) [inline Gemfile]: Note to self: change this to 2-2-dev

**Comment by athal7** [comment]
lgtm

File: lib/spree/wombat/handler/update_shipment_handler.rb [hunk 2]
```diff
@@ -107,7 +107,11 @@ def process
 
           #fire state transition
           if transition
-            shipment.fire_state_event(transition.event)
+            if target_state == 'shipped'
```
Comment (athal7) [inline lib/spree/wombat/handler/update_shipment_handler.rb]: do we need to also make sure the current state is not shipped? wouldn't want to ship it a second time unnecessarily

Comment (jordan-brough) [inline lib/spree/wombat/handler/update_shipment_handler.rb]: great point. we took a look and it looks like we're already covered by [this code](https://github.com/bonobos/spree_wombat/blob/99aa9031b4db7e871404095705bd35d940770ae5/lib/spree/wombat/handler/update_shipment_handler.rb#L97-L99) -- the `transition` will be nil if there isn't a transition to make. (if it's already in that state)

Comment (athal7) [inline lib/spree/wombat/handler/update_shipment_handler.rb]: also a great point.

**Comment by athal7** [comment]
:+1:

**Comment by athal7** [comment]
@philbirt all good here?

**Comment by philbirt** [comment]
Yep, I think we are good to go now. I'll merge shortly.

**Pull request merged**